VPLAY-9355 remove catch-up delay when processing a discontinuity

jameslofthouse-RED opened the pull request:
Do not wait for video to catch-up to audio if video is blocked processing a discontinuity

jameslofthouse-RED committed:
VPLAY-9355 Underflow causes aamp to go into pause state
Reason for change: If audio is processing a discontinuity, do not wait for it to catch up to video
Test Procedure: See ticket
Risks: Low

Signed-off-by: James Lofthouse <james_lofthouse@comcast.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=212) <br/>All committers have signed the CLA.

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9355_potential_delay_on_discontinuity

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
449fe91ba8e87de05bcf9e0fbc1c94b7
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->numberOfFragmentChunksCached" without holding lock "MediaTrack.mutex". Elsewhere, "MediaTrack.numberOfFragmentChunksCached" is written to with "MediaTrack.mutex" held 3 out of 3 times (2 of these accesses strongly imply that it is necessary).

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[streamabstraction.cpp:1840](undefined//blob/undefined/streamabstraction.cpp#L1840)

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9355_potential_delay_on_discontinuity

pstroffolino committed:
VPLAY-9355_potential_delay_on_discontinuity

Readson for Change: added 'override' directive to mocks to avoid compilation warning

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>
